packages/arb-validator/rollupvalidator/txtracker.go

```diff
@@ -229,6 +247,20 @@ func (tr *txTracker) processFinalizedAssertion(assertion rollup.FinalizedAsserti
 
 func (tr *txTracker) processRequest(request validatorRequest) {
 	switch request := request.(type) {
+	case outputMsgRequest:
+		assertionVal := newAssertionInfo()
+		for _, assertion := range tr.assertionInfo {
+			if request.assertionNodeHash == assertion.AssertNodeHash {
```

Comment:
It'd be good to add a separate mapping of nodeHash -> assertion so that we can do this efficiently